internal/consumer/partitionConsumer.go

```diff
@@ -256,6 +256,20 @@ func (p *partitionConsumer) messageLoop(offsetRange *kafka.OffsetRange) {
 	}
 }
 
+// delayMsg when Topic.Delay is not zero, it try to postpone the consumption of the msg for topic.Delay duration
+// if the current timestamp is too soon compare to the msg.Timestamp.
+func (p *partitionConsumer) delayMsg(timestamp time.Time) {
+	delay := p.topicPartition.Delay
+	// check non-zero msg delay
+	if delay > 0 {
+		sleepDuration := delay + timestamp.Sub(time.Now())
+		if sleepDuration > 0 {
+			p.logger.Debug("delay msg delivery", zap.Duration("sleepDuration", sleepDuration))
```

Comment:
Perhaps include message offset here also so we know which message is being delayed. You may need to pass in the full message instead of just `m.Timestamp` so that you can get `m.Offset`.